test/parallel/test-fs-read-stream-throw-type-error.js

```diff
@@ -25,3 +29,8 @@ createReadStreamErr(example, 123);
 createReadStreamErr(example, 0);
 createReadStreamErr(example, true);
 createReadStreamErr(example, false);
+
+// Should also throw on NaN (for https://github.com/nodejs/node/pull/19732)
+createReadStreamErr(example, { start: NaN });
+createReadStreamErr(example, { end: NaN });
+createReadStreamErr(example, { start: NaN, end: NaN });
```

Comment:
Please add some tests with `Infinity` as value.

Reply:
@addaleax Should the case `{start:0,end:Infinity}` throws (according to doc), or not (maybe for backward compatibility)?

Reply:
@anliting I think `Infinity` should be a valid value for `end`, yes… It should o the same as passing in `end: undefined` (or no `end:` value at all), but removing support for `Infinity` as a value is probably too much of a breaking change

Reply:
Exactly what I had been thinking. Doesn't throwing an error on `Infinity` make this `semver-major`?

Reply:
My appointment on that `{start:0,end:Infinity}` should throw is that the documentation does not allow passing `Infinity` as `end` (doc says `end` is an integer); no support is being removed, since it is never supported. In this point of view, "not throwing on `{start:0,end:Infinity}`" is a bug, and the old programs that worked are just lucky.

But passing `Infinity` as `end` is not harmful at all; it is even a good feature. Maybe we can change the doc, allowing it from now, making this a semver-minor?

Reply:
> But passing `Infinity` as `end` is not harmful at all; it is even a good feature.

:+1: 

> Maybe we can change the doc, allowing it from now, making this a semver-minor?

It’s probably a good idea to let the docs reflect this, but it might not even have to be semver-minor, if it’s just documenting existing behaviour. (After all, this is supposed to be a bug fix PR.)

Like, you’re right that technically we’d be adding support for something that is not documented as part of the API, but with Node we have to be conservative about these things and can’t just say “X is unsupported because it’s undocumented”.

Reply:
@addaleax Okay, now I understand your position for this, I'll get used to that. XD

Reply:
@addaleax so, should I also make changes to docs in this PR?

Reply:
@ryzokuken You can do that, or you can do it in another PR, or you can leave it up for somebody else to document this. But preferably one of the first two options :)

Reply:
option 2 sounds perfect. Let me address all the concerns regarding the code in this PR and I'd make another one for the documentation changes.